source/developers-guide/legacy-plugin-system/index.md

```diff
@@ -0,0 +1,820 @@
+
+class ControllerSubscriber extends SubscriberInterface
+{
+    /**
+     * @var Container
+     */
+    private $container;
+    
+    private $path;
+    
+    public function __construct(Container $container, $path) 
+    {
+        $this->container = $container;
+        $this->path = $path;
+    }
+    
+    /**
+     * @inheritdoc
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            'Enlight_Controller_Dispatcher_ControllerPath_Frontend_MyController' => 'registerController',
+        ];
+    }
+
+    public function registerController(\Enlight_Event_EventArgs $args)
+    {
+        $this->container->get('template')->addTemplateDir(
+            $this->getPath() . '/Views'
+        );
+
+        return $this->getPath() . '/Controllers/Frontend/MyController.php';
+    }
+}
+```
+Controller:
+```php
+<?php
+
+namespace Test\Controllers;
+
+class Shopware_Controllers_Frontend_MyController extends \Enlight_Controller_Action
+{
+    public function indexAction()
+    {
+    }
+}
+```
+
+### Frontend resources registration
+
+Additions to CSS, LESS and JavaScript resources had to be registered via `Theme_Compiler_Collect_Plugin_*` events. Since Shopware 5.2.13 the registration will be done automatically. Just place the resources into the following directories:
+
+* `custom/plugins/SwagResource/Resources/frontend/css/**.css`
+* `custom/plugins/SwagResource/Resources/frontend/js/**.js`
+* `custom/plugins/SwagResource/Resources/frontend/less/all.less`
+
+The `css` and `js` directories may contain arbitrary sub directories. `@import`s in `all.less` will be resolved.
+
+```php
+public function install()
+{
+    $this->subscribeEvent('Theme_Compiler_Collect_Plugin_Less', 'addLessFiles');
+    $this->subscribeEvent('Theme_Compiler_Collect_Plugin_Javascript', 'addJsFiles');
+}
+
+/**
+ * @return ArrayCollection
+ */
+public function addLessFiles()
+{
+    return new Doctrine\Common\Collections\ArrayCollection([
+        new LessDefinition(
+            [],
+            [__DIR__ . '/Views/frontend/_public/src/less/all.less'],
+            __DIR__
+        ),
+    ]);
+}
+
+/**
+ * @return ArrayCollection
+ */
+public function addJsFiles()
+{
+    return new ArrayCollection([
+        __DIR__ . '/Views/frontend/_public/src/js/jquery.swag_live_shopping.js',
+    ]);
+}
+```
+
+## Add console commands
+For register a console command create a command file which extends the class "ShopwareCommand".
+The register the event `Shopware_Console_Add_Command` and return a new ArrayCollection with your created commands.
+
+```php
+$this->subscribeEvent(
+    'Shopware_Console_Add_Command',
+    'onAddConsoleCommand'
+);
+
+/**
+ * Adds the console commands
+ *
+ * @return ArrayCollection
+ */
+public function onAddConsoleCommand()
+{
+    return new ArrayCollection(
+        [
+          new Command1(),
+          new Command2(),
+        ]
+    );
+}
+```
+
+```php
+<?php
+
+class ExportCommand extends ShopwareCommand
```

Comment:
should be named Command1 as you use this name as example above

Reply:
fd